File: TheAnvil/src/main/java/com/qualityplus/anvil/base/TheAnvilAPIImpl.java
package com.qualityplus.anvil.base;

import com.qualityplus.anvil.api.TheAnvilAPI;
import com.qualityplus.anvil.api.provider.EnchantmentProvider;
import com.qualityplus.assistant.lib.eu.okaeri.injector.annotation.Inject;
import com.qualityplus.assistant.lib.eu.okaeri.platform.core.annotation.Component;
import lombok.Getter;

@Component
public final class TheAnvilAPIImpl implements TheAnvilAPI {
    private @Inject
    @Getter EnchantmentProvider provider;
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/base/TheAnvilAPIImpl.kt
package com.qualityplus.anvil.base

import com.qualityplus.anvil.api.TheAnvilAPI
import com.qualityplus.anvil.api.provider.EnchantmentProvider
import eu.okaeri.injector.annotation.Inject
import eu.okaeri.platform.core.annotation.Component
import lombok.Getter

@Component
class TheAnvilAPIImpl : TheAnvilAPI {
    @Inject
    @Getter
    private override val provider: EnchantmentProvider? = null
}